addon/-private/system/relationships/state/has-many.js

```diff
@@ -112,6 +112,8 @@ export default class ManyRelationship extends Relationship {
     if (this._manyArray) {
       this._manyArray.flushCanonical();
     }
+
+    this.notifyHasManyChanged();
```

Comment:
No - the array hasn't necessarily changed - the above code is there to identify whether anything has changed and to notify the minimum possible change.

This addition should not be made. It nullifies the optimisation.